File: examples/profile-card.ts
/**
 * Example: Profile Card View
 *
 * A SwiftUI view that shows a user's profile with an
 * expandable details section. Demonstrates props, state,
 * and conditional rendering.
 *
 * Run:
 *   axint compile examples/profile-card.ts --out generated/
 */

import { defineView, prop, state, view } from "@axint/compiler";

export default defineView({
  name: "ProfileCard",
  props: {
    displayName: prop.string("User's display name"),
    joinDate: prop.date("Account creation date"),
    avatarURL: prop.url("Profile image URL"),
  },
  state: {
    showDetails: state.boolean("Whether details are visible", { default: false }),
  },
  body: [
    view.vstack([
      view.hstack([
        view.text("entry.displayName"),
      ], { spacing: 12 }),
      view.button("showDetails ? \"Hide Details\" : \"Show Details\"", "showDetails.toggle()"),
      view.conditional("showDetails", [
        view.text("entry.joinDate"),
      ]),
    ], { alignment: "leading", spacing: 12 }),
  ],
});

File: python/examples/profile_card.py
"""Example: profile card view.

Run it:

    axint-py parse examples/profile_card.py
    axint-py compile examples/profile_card.py --stdout
"""

from axint import define_view, prop, state, view

profile_card = define_view(
    name="ProfileCard",
    props={
        "display_name": prop.string("User display name"),
        "join_date": prop.date("Join date"),
        "avatar_url": prop.url("Avatar URL"),
    },
    state={
        "show_details": state.boolean("Show details", default=False),
    },
    body=[
        view.vstack(
            [
                view.text("display_name"),
                view.button("Toggle details", "show_details.toggle()"),
                view.conditional(
                    "show_details",
                    [
                        view.text("join_date"),
                    ],
                ),
            ],
            spacing=12,
            alignment="leading",
        )
    ],
)
